src/core/utils/i18n.js

```diff
@@ -39,3 +40,83 @@ export const getLocalizedTextWithLinkParts = ({
     afterLinkText: parts[2],
   };
 };
+
+const getReplacementKey = (start: string, end: string): string => {
+  return `${start},${end}`;
+};
+
+type ReplaceStringsWithJSXParams = {|
+  text: string,
+  replacements: Array<[string, string, (text: string) => React.Node]>,
+|};
+
+export const replaceStringsWithJSX = ({
+  text,
+  replacements,
+}: ReplaceStringsWithJSXParams): Array<React.Node | string> => {
+  if (replacements.length === 0) {
```

Comment:
Is there a realistic case for this? If not, maybe it should be an error since it could be a programming mistake.